web/packages/teleterm/src/mainProcess/mainProcessClient.ts

```diff
@@ -199,5 +202,40 @@ export default function createMainProcessClient(): MainProcessClient {
         args
       );
     },
+    checkForAppUpdates() {
+      return ipcRenderer.invoke(MainProcessIpc.CheckForAppUpdates);
+    },
+    downloadAppUpdate() {
+      return ipcRenderer.invoke(MainProcessIpc.DownloadAppUpdate);
+    },
+    cancelAppUpdateDownload() {
+      return ipcRenderer.invoke(MainProcessIpc.CancelAppUpdateDownload);
+    },
+    quitAndInstallAppUpdate() {
+      return ipcRenderer.invoke(MainProcessIpc.QuiteAndInstallAppUpdate);
+    },
+    changeAppUpdatesManagingCluster(clusterUri) {
+      return ipcRenderer.invoke(
+        MainProcessIpc.ChangeAppUpdatesManagingCluster,
+        {
+          clusterUri,
+        }
+      );
+    },
+    subscribeToAppUpdateEvents: listener => {
+      const ipcListener = (_, updateEvent: AppUpdateEvent) => {
+        // Deserialize the error.
+        if (updateEvent.kind === 'error') {
+          updateEvent.error = ensureError(updateEvent.error);
```

Comment:
If you control both ends (one that serializes the error and one that deserializes it), I imagine in a perfect world it'd be better to have a specialized function for this which handles the exact serialization shape used by the other end. Depending on `ensureError` to do this job feels a bit off but I think it's acceptable. ;f

Reply:
I added functions to serialize/deserialize errors. I got rid of serialization in `appUpdater` (which doesn't need to be aware of it), now this happens before sending the event through IPC.